src/rules/javascript/content-scripts-file-absent.js

```diff
@@ -39,12 +40,24 @@ export default {
             });
             return;
           }
-          let normalizedName = path.resolve(dirname, fileValue);
-          if (path.isAbsolute(fileValue)) {
-            normalizedName = path.join(path.resolve('.'), path.normalize(fileValue));
+
+          // We can't reliably validate relative file names because their
+          // behaviour depends on the current page and the behaviour for
+          // Firefox and Chrome differentiates too. So we chose only to
+          // validate absolute file paths to avoid false positives.
+          if (!path.isAbsolute(fileValue)) {
+            return;
           }
+
+          const normalizedName = path.join(
```

Comment:
I'm wondering if now that we are only going to validate absolute paths (and so we are sure that the file name is going to start with '/') we could simplify this checks by just resolving the file as relative to '/', e.g.:

```js
path.relative("/", "/dir/file.js") // returns "dir/file.js"

path.relative("/", "/dir/../dir/./file.js") // returns "dir/file.js"

path.relative("/", "/dir/../../../.././file.js") // returns "file.js"
```

and then just check if the file is in the existingFileName map without any need to convert them info absolute urls.
(I also double-checked that what we do internally in Firefox is basically the same kind of resolution here: https://searchfox.org/mozilla-central/rev/877c99c523a054419ec964d4dfb3f0cadac9d497/toolkit/components/extensions/ext-tabs-base.js#661)

This way we could simplify this to something like:

```js
const normalizedName = path.resolve('/', fileValue);

if (Object.keys(existingFiles).includes(normalizedName)) {
  ...
}
```

which seems also easier to read.

@EnTeQuAk How that sounds to you?

Reply:
That's tons cleaner. I'll update the rule :smiley:

Reply:
That worked mostly, I still need to normalize the `existingFiles` object which is why I kept it but we're now resolving everything to `'/'` and avoid the explicit join on `dirname`.

That already looks a bit better.

I wasn't able to not normalize the `existingFiles` because they're all relative paths so we have to make them absolute which seemed like the more reasonable approach.